shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLoggerImpl.java

```diff
@@ -95,6 +95,24 @@ public Counter call() throws Exception {
     }
 
     @Override
+    public long getCounterValue(String name) {
+        Exceptions.checkNotNullOrEmpty(name, "name");
+        String counterName = name + ".Counter";
+        try {
+            Counter counter = countersCache.get(counterName, new Callable<Counter>() {
+                @Override
+                public Counter call() throws Exception {
+                    return underlying.createCounter(counterName);
```

Comment:
IMO get should not have a sideeffect on metadata.. this is cache we are talking about.. in case of cache miss, get should retrieve and load the value.. the important distinction to note is that this get is not updating the metadata anywhere for this get to have any side effect..